### ISSUE TITLE
Question about point_object_class label

### ISSUE THREAD

**Issue by WeitaiKang** [nickgkan/beauty_detr#47]
Dear authors, 

Thank you so much for your influential work. Your patience in solving so many issues do help a lot new researcher in this 3D field. As a new researcher in this field, i may want to ask a little basic question which i don't understand but find it widely-used case.

In the [compute_points_obj_cls_loss_hard_topk](https://github.com/nickgkan/butd_detr/blob/10570e0b6826d4a236b18c2c8fac5903866e1c60/models/losses.py#L161) function, i find that you assign the object class label (binary value) to just the topk closest points among all the object points and the background points. Such kind of result (**objectness_label**) serves as the ground truth for model's **seeds_obj_cls_logits**.

I am confuse about why we only treat the topk closest point as the object for **seeds_obj_cls_logits**, instead of just all the object points (means directly use **obj_assignment_one_hot** as the ground truth). Is it that because in the loss computation, we only treat one proposal to calculate loss of one target (assigned by **HungarianMatcher**)? If so, why top-k, instead of top-1?

**Comment by ayushjain1144** [comment]
Hi,

this code is trying to match the predicted box centers to ground truth
object centers. matching only the closest predicted to ground truth and
supervising the rest as background can be too harsh on some proposals which
are reasonably near to the prediction. letting all objects to the center
would wrongly let bad predictions to survive. topk helps to have a balance
between these extremes.

however, note that this is pretty empirical and works which follow DETR
style matching just do 1NN matching (but they take into account multiple
factors simultaneously for matching). we took the above design from
group-free.

**Comment by WeitaiKang** [comment]
Thank you so much for your detailed explanation. I appreciate the time you took to clarify these concepts.
